### DISCUSSION TITLE
how to account for drift in the initial hovering

### DISCUSSION THREAD

**Discussion by iNavin** [bitcraze/discussions#983] [Q&A] [answered]
hi, 

I am trying to perform path planning over a graph with standard path planning algorithm( i am using flow deck for positioning), but the drone is drifting to a different position,
can you suggest some way or some resources(examples) to minimize this drift so that crazyflie can follow the intended path.

for example :
<img width="571" alt="image" src="https://github.com/bitcraze/discussions/assets/41445769/0d5a4492-eb18-4fdb-ac8c-8e754213a5d0">

the blue line represents the ideal trajectory, and the red line represents the trajectory that drone follows.

**Comment by krichardsson** [comment, marked as answer]
It looks like there is an offset in x and y. 

The flow deck (or actually the estimator) does not know anything about absolute position as it can only measure changes in X, Y. 
The estimator tries to continuously track the absolute position based on the available sensor input and uses the data from the flow deck to 
add the deltas to the current estimated position. 

It looks like your starting position is (0, 0) which indicates that you reset the estimator just before takeoff. If you do not reset the estimator it will have some random position due to the noise in the IMU. 

If you are not taking off from (0, 0) and you want to reset the estimator to this position instead (I think this is actually what you are looking for), just set the [kalman.initialX](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/params/#kalmaninitialx) and [kalman.initialY](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/params/#kalmaninitialy) before resetting the estimator (by setting [kalman.resetEstimation](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/params/#kalmanresetestimation))

**Comment by iNavin** [comment]
yes this worked thanks for your suggestion, 
![output 3](https://github.com/bitcraze/discussions/assets/41445769/31b25a97-5711-40c5-9347-6136fb5bfdf5)

now the drift has also reduced.
